### DISCUSSION TITLE
Implementing Horizontal pod autoscaling in Rabbitmq cluster deployed using CRDsn

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#8893] [Questions] [answered]
I have deployed a Rabbitmq cluster in AKS using CRD. Now I am trying to implement HPA in rabbitmq to upscale and downscale as needed, but not getting successful in rabbitmq kind (pod).

Does anyone have any suggestion/idea on how to implement horizontal pod autoscaling in Rabbitmq cluster?

Deployed in Aks- Rabbitmq cluster kubernetes operator, messaging topology operator, rabbitmq cluster. Rabbitmq image is official and not bitnami one.

**Comment by michaelklishin** [comment]
There should be existing Kubernetes tools that use a metric or a function and can scale a set of pods. I would not recommend any specific ones.

The hard part is not picking the tool, or even developing one. The hard part is understanding your workload, and whether adding nodes would make *any* difference for it.

As the most obvious example, if all your clients connect to one node and all queue/stream leader replicas are on the same node, adding more node won't change a thing.

**Reply by mkuratczyk** [reply, marked as answer]
It's even easier than that - the Operator explicitly doesn't support scaling down: https://github.com/rabbitmq/cluster-operator/issues/223, so it just can't be done in this setup.

But as Michael said, the most interesting part is what do you think you can accomplish by scaling up and down @PayalMetange.

**Reply by paywagh11** [reply]
Thank you so much for your answers. I will try to understand my use case more thoroughly and will get back to you in case of any concerns.
